plugins/woocommerce/src/Internal/DataStores/Orders/DataSynchronizer.php

```diff
@@ -171,14 +185,123 @@ public function delete_database_tables() {
 	}
 
 	/**
-	 * Is the data sync between old and new tables currently enabled?
+	 * Is the real-time data sync between old and new tables currently enabled?
 	 *
 	 * @return bool
 	 */
 	public function data_sync_is_enabled(): bool {
 		return 'yes' === get_option( self::ORDERS_DATA_SYNC_ENABLED_OPTION );
 	}
 
+	/**
+	 * Is the background data sync between old and new tables currently enabled?
+	 *
+	 * @return bool
+	 */
+	public function background_sync_is_enabled(): bool {
+		$enabled = $this->data_sync_is_enabled();
+
+		/**
+		 * Toggle for background sync.
+		 *
+		 * This allows for background sync to be enabled even when real-time sync is disabled.
+		 *
+		 * @param bool $enabled True to enabled background sync.
+		 */
+		return apply_filters( 'woocommerce_custom_orders_table_background_sync_enabled', $enabled );
+	}
+
+	/**
+	 * Schedule an event to run background sync when the mode is not set to continuous.
+	 *
+	 * @return void
+	 */
+	private function schedule_background_sync() {
+		$has_scheduled_action = WC()->queue()->search( array(
```

Comment:
This an expensive call to do on every init, we need to either cache it or link it to a less frequent function. Probably both. wdyt about linking to admin_init instead of init, and then storing the epoch time in an option when this was last checked, and checking only if last_checked was more than 1 hour ago?

Reply:
I think it's important for`handle_background_sync` to run on `init`. If it only runs on admin requests, it could be a long time between runs of that callback, and in the mean time a lot of changes to orders could happen on the front end.

I added transient caching in e63ea3bf71fc32e3f40bad4da103dbdbdd17e273, let me know what you think of that.

Reply:
Noting that there is an [open issue](https://github.com/woocommerce/action-scheduler/issues/978) in Action Scheduler to improve the performance of this particular query.